### DISCUSSION TITLE
How to set up the various external oauth providers (Github, MS, Google, etc.)

### DISCUSSION THREAD

**Discussion by AntonOfTheWoods** [fief-dev/fief#277] [Q&A] [answered]
`Auth WarningAuthorization may be unsafe, passed state was changed in server. The passed state wasn't returned from auth server.`

I have been getting these when trying to set up a test GH login. I originally failed to do this in docker compose and am now failing with the cloud hosted beta.

```
Auth Error[redirect_uri_mismatch]: The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.. More info: https://docs.github.com/apps/managing-oauth-apps/troubleshooting-authorization-request-errors/#redirect-uri-mismatch
```

Are there any instructions on how to do this? I have also been getting: `error: invalid_grant` in another session but when starting from scratch in a private browsing session I am getting the above errors.

Are there any walkthroughs of how to set this up with various external providers, at least with the cloud hosted beta?

**Comment by frankie567** [comment, marked as answer]
You need to set the Redirect URI on your GitHub application settings. The steps are detailed here: https://docs.fief.dev/configure/oauth-providers/#github

**Reply by AntonOfTheWoods** [reply]
Ok, sorry, don't know how I missed that, I was creating an oauth app rather than a github app.
